src/asmcnc/skavaUI/screen_go.py

```diff
@@ -484,32 +490,41 @@ def on_pre_enter(self, *args):
         else:
             self.reset_go_screen_prior_to_job_start()
 
+        if self.show_maintenance_prompts() and use_sc2: self.get_sc2_brush_data()
+
     def on_enter(self):
 
-        # Check for SC2 compatability
-        if self.m.is_machines_fw_version_equal_to_or_greater_than_version('2.2.8', 'Spindle lifetime check') \
-            and self.m.theateam() and self.m.get_dollar_setting(51) and self.m.stylus_router_choice != 'stylus':
-            sc2_compatible = True
+        if not self.getting_brush_data and self.show_maintenance_prompts():
```

Comment:
I think this should also have the condition `not self.m.is_using_sc2()`

Reply:
Also, the `self.getting_brush_data` doesn't seem to have a use as far as I can see - I would move this to on_pre_enter and rewrite this as:
`if self.show_maintenance_prompts():
    if use_sc2: self.get_sc2_brush_data()
    else: check_brush_use_and_lifetime`
Let me know if I'm missing something here

Reply:
For some reason I thought the popups had to be called on_enter, but I have no idea why.

Anyway, moved everything into pre-enter and retested :). Please re-review.

Thank you!